app/models/concerns/cohort_ongoing_enrollments.rb

```diff
@@ -18,10 +18,13 @@ def display_read_only(user)
   private def for_display(column, user)
     return nil unless cohort_client.client.processed_service_history&.public_send(column)
 
+    enforce_visibility = cohort_client.cohort.enforce_project_visibility_on_cells?
     # in the form [{project_name: 'Project Name', date: 'last date', project_id: 'Project ID}]
     cohort_client.client.processed_service_history.public_send(column).
       select do |row|
-        row['project_id'].in? user.visible_project_ids_enrollment_context
+        next true unless enforce_visibility
```

Comment:
Needs to return projects that are visible in the window if `cohort.only_window` otherwise true.